src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/links/LinkInfo.java

```diff
@@ -125,7 +151,10 @@ public Content getClassLinkLabel(BaseConfiguration configuration) {
             return label;
         } else if (isLinkable()) {
             Content tlabel = newContent();
-            tlabel.add(configuration.utils.getSimpleName(typeElement));
+            tlabel.add(type instanceof DeclaredType dt && linkEnclosingTypes(dt)
```

Comment:
There's a philosophical discussion to be had, separately, with Joe, about using `instanceof` with Language Model types. since the beginning, i`nstanceof` has been discouraged in favor of using `getKind()` but the reasons for that may have faded, and "instanced patterns" may be enough of a reason to change the guidelines.